File: app/src/main/java/igrek/forceawaken/domain/alarm/AlarmsConfig.java
package igrek.forceawaken.domain.alarm;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;

public class AlarmsConfig implements Parcelable {
	
	private ArrayList<AlarmTrigger> alarmTriggers = new ArrayList<>();
	
	public AlarmsConfig() {
	}
	
	public ArrayList<AlarmTrigger> getAlarmTriggers() {
		return alarmTriggers;
	}
	
	@Override
	public int describeContents() {
		return 0;
	}
	
	@Override
	public void writeToParcel(Parcel dest, int flags) {
		dest.writeList(alarmTriggers);
	}
	
	private AlarmsConfig(Parcel in) {
		this.alarmTriggers = in.readArrayList(AlarmTrigger.class.getClassLoader());
	}
	
	public static final Parcelable.Creator<AlarmsConfig> CREATOR = new Parcelable.Creator<AlarmsConfig>() {
		public AlarmsConfig createFromParcel(Parcel in) {
			return new AlarmsConfig(in);
		}
		
		public AlarmsConfig[] newArray(int size) {
			return new AlarmsConfig[size];
		}
	};
}

File: app/src/main/java/igrek/forceawaken/alarm/AlarmsConfig.kt
package igrek.forceawaken.alarm

import android.os.Parcel
import android.os.Parcelable


data class AlarmsConfig(
    val alarmTriggers: MutableList<AlarmTrigger> = mutableListOf(),
    val repetitiveAlarms: MutableList<RepetitiveAlarm> = mutableListOf(),
) : Parcelable {

    companion object {
        @JvmField
        val CREATOR = object : Parcelable.Creator<AlarmsConfig> {
            override fun createFromParcel(parcel: Parcel): AlarmsConfig {
                val alarmTriggers =
                    parcel.readArrayList(AlarmTrigger::class.java.classLoader) as List<AlarmTrigger>
                val repetitiveAlarms =
                    parcel.readArrayList(RepetitiveAlarm::class.java.classLoader) as List<RepetitiveAlarm>
                return AlarmsConfig(alarmTriggers.toMutableList(), repetitiveAlarms.toMutableList())
            }

            override fun newArray(size: Int) = arrayOfNulls<AlarmsConfig>(size)
        }
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeList(alarmTriggers.toList())
        dest.writeList(repetitiveAlarms.toList())
    }

    override fun describeContents() = 0
}
